assets/js/system-status.ts

```diff
@@ -93,49 +82,36 @@ window.addEventListener('DOMContentLoaded', () => {
     fetch('https://status.maxmind.com/1.0/status/53fcfbb2ac0c957972000235')
       .then((res) => res.json())
       .then((json) => {
-        const systemStatusCode = json.result.status_overall.status_code;
-        switch (systemStatusCode) {
-          case 100:
-            setSystemStatus(
-              status.OPERATIONAL.icons,
-              status.OPERATIONAL.message
-            );
-            break;
-          case 200:
-            setSystemStatus(
-              status.PLANNED_MAINTENANCE.icons,
-              status.PLANNED_MAINTENANCE.message
-            );
-            break;
-          case 300:
-            setSystemStatus(
-              status.DEGRADED_PERFORMANCE.icons,
-              status.DEGRADED_PERFORMANCE.message
-            );
-            break;
-          case 400:
-            setSystemStatus(
-              status.PARTIAL_SERVICE_DISRUPTION.icons,
-              status.PARTIAL_SERVICE_DISRUPTION.message
-            );
-            break;
-          case 500:
-            setSystemStatus(
-              status.SERVICE_DISRUPTION.icons,
-              status.SERVICE_DISRUPTION.message
-            );
-            break;
-          case 600:
-            setSystemStatus(
-              status.SECURITY_EVENT.icons,
-              status.SECURITY_EVENT.message
-            );
-            break;
+        const status_code = json.result.status_overall.status_code;
+        if (!(status_code in status)) {
+          throw new TypeError('status_code invalid');
         }
+        if (json.result.incidents.length != 0) {
+          setSystemStatus({
+            icon: $warningIcons,
+            message: json.result.incidents[0].name,
+            title: status[Number(status_code)].title,
+          });
+          return;
+        } else if (json.result.maintenance.active.length != 0) {
+          setSystemStatus({
+            icon: $warningIcons,
+            message: json.result.maintenance.active[0].name,
+            title: status[Number(status_code)].title,
+          });
+          return;
+        }
+        setSystemStatus(status[Number(status_code)]);
       })
       .catch(() => {
-        throw new Error('There was an error retrieving the status.');
+        /**
+         * No-op
+         *
+         * If something goes wrong, we intentionally want to swallow the error
+         * and prevent the UI from knowing
+         */
```

Comment:
I'm not 100% sure about this. It brings it in line with the dev-site behaviour, but it does potentially paper over a real problem.

Reply:
You could always do `console.error(error)` here. That way it would show up in the console.

Reply:
I think someone else brought this up before too. @kevcenteno @mm-jpoole, thoughts?